include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -77,8 +120,9 @@ class device_policy
         return ::std::true_type{};
     }
 
-  private:
-    sycl::queue q;
+  protected:
+    mutable ::std::optional<sycl::queue> q;
+    mutable ::std::mutex mtx;
```

Comment:
From my point of view better to declare synchronization staff at first.

Reply:
Done.